src/components/slider/slider.tsx

```diff
@@ -0,0 +1,433 @@
+  lowerLimit = 0,
+  upperLimit = 100,
+  onValueChange,
+  isRange = false,
+  showEndMarkers = false,
+  internalMarkerStep,
+  showValueInput = false,
+  disabled = false,
+}: SliderProps) => {
+  const { sizes } = useTheme();
+  const handleSize = Number(sizes['5']);
+  const [trackLength, setTrackLength] = useState(0);
+
+  const styles = useSliderStyles();
+
+  const calculatePositionFromValue = (val: number) =>
+    trackLength > 0 ? ((val - lowerLimit) / (upperLimit - lowerLimit)) * trackLength : 0;
+
+  const calculateValueFromPosition = (pos: number) =>
+    trackLength > 0 ? lowerLimit + ((upperLimit - lowerLimit) * pos) / trackLength : lowerLimit;
+
+  const applyStep = (val: number) => {
+    if (step > 0) {
+      const stepped = Math.round(val / step) * step;
+      return Math.min(Math.max(stepped, lowerLimit), upperLimit);
+    }
+    return val;
+  };
+
+  const internalMarkers = useMemo(() => {
+    if (!internalMarkerStep || internalMarkerStep <= 0) return [];
+    const markers = [];
+    for (let i = lowerLimit + internalMarkerStep; i < upperLimit; i += internalMarkerStep) {
+      markers.push(i);
+    }
+    return markers;
+  }, [internalMarkerStep, lowerLimit, upperLimit]);
+
+  const initialRange: [number, number] =
+    Array.isArray(value) && isRange ? value : [lowerLimit, upperLimit];
+
+  const initialValue: number = typeof value === 'number' && !isRange ? value : lowerLimit;
+
+  const [range, setRange] = useState<[number, number]>(initialRange);
+  const [sliderValue, setSliderValue] = useState(initialValue);
+  const [inputMin, setInputMin] = useState(isRange ? `${initialRange[0]}` : `${initialValue}`);
+  const [inputMax, setInputMax] = useState(`${initialRange[1]}`);
+  const [isSliding, setIsSliding] = useState(true);
+  const [bubbleValue, setBubbleValue] = useState<number | null>(null);
+  const [activeHandle, setActiveHandle] = useState<BubbleHandle | null>(null);
+
+  const rangeLeftPos = useRef(
+    new Animated.Value(calculatePositionFromValue(initialRange[0])),
+  ).current;
+  const rangeRightPos = useRef(
+    new Animated.Value(calculatePositionFromValue(initialRange[1])),
+  ).current;
+  const handlePosition = useRef(
+    new Animated.Value(calculatePositionFromValue(initialValue)),
+  ).current;
+
+  const lastLeft = useRef(0);
+  const lastRight = useRef(0);
+  const lastPosition = useRef(0);
+
+  useEffect(() => {
+    if (trackLength > 0) {
+      const left = calculatePositionFromValue(range[0]);
+      const right = calculatePositionFromValue(range[1]);
+      const pos = calculatePositionFromValue(sliderValue);
+
+      rangeLeftPos.setValue(left);
+      rangeRightPos.setValue(right);
+      handlePosition.setValue(pos);
+
+      lastLeft.current = left;
+      lastRight.current = right;
+      lastPosition.current = pos;
+    }
+  }, [trackLength]);
+
+  useEffect(() => {
+    if (trackLength <= 0) return;
+    if (trackLength > 0) {
+      if (isRange && Array.isArray(value)) {
```

Comment:
The code is extremely hard to follow through.

1. Add code comments with example so one can easily understand it.
2. Consider breaking into smaller functions with appropriate semantic naming.